src/java/org/apache/cassandra/db/compaction/TableOperation.java

```diff
@@ -63,6 +58,12 @@ public interface TableOperation
      */
     boolean isStopRequested();
 
+    default void throwIfStopRequested()
```

Comment:
I have the impression that the motivation for adding this method to the interface is to deduplicate the code rather than the necessity to extend the capability of the interface. It's probably nothing major but I wanted to leave this comment as an observation.

Reply:
Yes, it's just a utility method since there are 11 calls using it. We could put it in `AbstractTableOperation` but there are some usages that only get access to the interface.